Yosemite/Yosemite/PointOfSale/POSProductProvider.swift

```diff
@@ -48,14 +49,15 @@ public final class POSProductProvider: POSItemProvider {
             DDLogError("Error fetching products from storage")
         }
 
-        // 2. Map result to POSProduct and populate the output
+        // 2. Map result to POSProduct, and populate the output already formatted with any applicable store settings
+        let currencyFormatter = CurrencyFormatter(currencySettings: currencySettings)
         return loadedProducts.map { product in
-            // TODO: Decorate the price with currency formatting
-            POSProduct(itemID: UUID(),
-                       productID: product.productID,
-                       name: product.name,
-                       price: product.price,
-                       formattedPrice: "$\(product.price)")
+            let formattedPrice = currencyFormatter.formatAmount(product.price, with: currencySettings.currencyCode.rawValue) ?? "-"
```

Comment:
The `-` nil-coalescing fallback can be improved when handling unhappy paths. I'm not quite sure right now what we want to display if there's a failure converting the price.